server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2144,6 +2146,8 @@ private IndexWriterConfig getIndexWriterConfig() {
         final IndexWriterConfig iwc = new IndexWriterConfig(engineConfig.getAnalyzer());
         iwc.setCommitOnClose(false); // we by default don't commit on close
         iwc.setOpenMode(IndexWriterConfig.OpenMode.APPEND);
+        iwc.setReaderAttributes(Collections.singletonMap(BlockTreeTermsReader.FST_MODE_KEY + "." + IdFieldMapper.NAME,
+            BlockTreeTermsReader.FSTLoadMode.ON_HEAP.name())); // always force ID field on-heap for fast updates
```

Comment:
should we also set `FST_MODE_KEY` to `OFF_HEAP` so that other fields than `_id` remain off-heap?

Reply:
hmm I see, I was coming from a different angle which was to avoid loading stuff on-heap for unique fields that are not the `_id`

Reply:
I pushed a new commit as a suggestion. I need to add a test for it too.